worker/credentialvalidator/worker_test.go

```diff
@@ -78,60 +92,78 @@ func (s *WorkerSuite) TestModelCredentialNotNeeded(c *gc.C) {
 	s.facade.CheckCallNames(c, "ModelCredential")
 }
 
-func (s *WorkerSuite) TestCredentialChangeToInvalid(c *gc.C) {
-	s.facade.credentials = []base.StoredCredential{
-		{credentialTag, true},
-		{credentialTag, false},
-	}
+func (s *WorkerSuite) TestModelCredentialUnsetWhileRunning(c *gc.C) {
+	worker, err := testWorker(s.config)
+	c.Assert(err, jc.ErrorIsNil)
+
+	s.sendChange(c)
```

Comment:
Is the first change needed? This should still pass with exists=false followed by sendChange()?
Same comment for other test cases below.